packages/odata-service-writer/src/index.ts

```diff
@@ -11,24 +11,51 @@ import { t } from './i18n';
 import { OdataService, OdataVersion } from './types';
 
 /**
- * Validates the provided base path.
+ * Validates the existence of the given files in the provided base path.
+ *
+ * @param basePath - the root path of an existing UI5 application
+ * @param files - list of files that need to exist
+ * @param fs - the memfs editor instance
+ */
+function validateBasePath(basePath: string, files: string[], fs: Editor) {
+    files.forEach((path) => {
+        if (!fs.exists(join(basePath, path))) {
+            throw new Error(t('error.requiredProjectFileNotFound', { path }));
+        }
+    });
+}
+
+/**
+ * Try finding a package.json and a ui5.yaml for the given project by looking upwards in the folder hierachy.
  *
  * @param {string} basePath - the root path of an existing UI5 application
- * @param {Editor} fs - the memfs editor instance
+ * @param {Editor} [fs] - the memfs editor instance
+ * @returns an object with the optional locations of the package.json and ui5.yaml
  */
-function validateBasePath(basePath: string, fs: Editor) {
-    [join(basePath, 'package.json'), join(basePath, 'webapp', 'manifest.json'), join(basePath, 'ui5.yaml')].forEach(
-        (path) => {
-            if (!fs.exists(path)) {
-                throw new Error(t('error.requiredProjectFileNotFound', { path }));
-            }
+export async function findProjectFiles(
+    basePath: string,
+    fs: Editor
+): Promise<{ packageJson?: string; ui5Yaml?: string }> {
+    const files: { packageJson?: string; ui5Yaml?: string } = {};
+    const parts = join(basePath).split('/');
+
+    while (parts.length > 0 && (!files.packageJson || !files.ui5Yaml)) {
```

Comment:
Does it make sense to use a library like [`find-up`](https://github.com/sindresorhus/find-up) to do this? The library will take care things like path normalization and handle the various platforms (Windows path idiosyncrasies come to mind!). We make extensive use of `find-up` in tools-suite.

Reply:
I tried using it but `find-up` works with the real filesystem while we work with `mem-fs`. I tried using directory callbacks in `find-up` to handle `mem-fs` but it only partially worked for me and got so complicated that doing it on my own seemed simpler (and worked).
If you want to give it a short, please try, I am happy to use `find-up` if we get it running with `mem-fs`.

Reply:
PS: now it works on windows as well ;) (https://github.com/SAP/open-ux-tools/pull/501/commits/783997f0bb83e64ca955f4c0159f14463fea4dcc)